src/server/game/Grids/GridStates.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
```

Comment:
https://www.azerothcore.org/wiki/file-headers

Also please find some referenced commits from TC so we can properly credit the author(s)

Reply:
Oh ye ofc. I went ahead a checked commit history for GridStates.cpp, GridStates.h and NGrid.cpp, however it seems like all of them were just added out of the blue, way back in 2008 or something. And very little has changed over time. Someone called Megamage seems to have done most of the changes but his account no longer exists, so idk.

Reply:
Then it is mostlikely from Mangos era

Reply:
Ye true